src/tests/ftest/util/test_utils.py

```diff
@@ -0,0 +1,928 @@
+                self.log.error(msg)
+        return check_status
+
+    def rebuild_complete(self):
+        """Determine if the pool rebuild is complete.
+
+        Returns:
+            bool: True if pool rebuild is complete; False otherwise
+
+        """
+        self.get_info()
+        return self.info.pi_rebuild_st.rs_done == 1
+
+    def wait_for_rebuild(self, to_start, interval=1):
+        """Wait for the rebuild to start or end.
+
+        Args:
+            to_start (bool): whether to wait for rebuild to start or end
+            interval (int): number of seconds to wait in between rebuild
+                completion checks
+        """
+        self.log.info(
+            "Waiting for rebuild to %s ...",
+            "start" if to_start else "complete")
+        while self.rebuild_complete() == to_start:
+            self.log.info(
+                "  Rebuild %s ...",
+                "has not yet started" if to_start else "in progress")
+            sleep(interval)
+        self.log.info(
+            "Rebuild %s detected", "start" if to_start else "completion")
+
+    @fail_on(DaosApiError)
+    def start_rebuild(self, server_group, rank, daos_log):
+        """Kill a specific server rank using this pool.
+
+        Args:
+            server_group (str): daos server group name
+            rank (int): daos server rank to kill
+            daos_log (DaosLog): object for logging messages
+
+        Returns:
+            bool: True if the server has been killed and the rank has been
+            excluded from the pool; False if the pool is undefined
+
+        """
+        msg = "Killing DAOS server {} (rank {})".format(server_group, rank)
+        self.log.info(msg)
+        daos_log.info(msg)
+        server = DaosServer(self.context, server_group, rank)
+        server.kill(1)
+        return self.exclude(rank, daos_log)
+
+    @fail_on(DaosApiError)
+    def exclude(self, rank, daos_log):
+        """Manually exclude a rank from this pool.
+
+        Args:
+            rank (int): daos server rank to kill
+            daos_log (DaosLog): object for logging messages
+
+        Returns:
+            bool: True if rank has been excluded from the pool; False if the
+                pool is undefined
+
+        """
+        if self.pool:
+            msg = "Excluding server rank {} from pool {}".format(
+                rank, self.uuid)
+            self.log.info(msg)
+            daos_log.info(msg)
+            self.pool.exclude([rank])
+            return True
+        return False
+
+    def check_files(self, hosts):
+        """Check if pool files exist on the specified list of hosts.
+
+        Args:
+            hosts (list): list of hosts
+
+        Returns:
+            bool: True if the files for this pool exist on each host; False
+                otherwise
+
+        """
+        return check_pool_files(self.log, hosts, self.uuid.lower())
+
+    def write_file(self, orterun, processes, hostfile, size, timeout=60):
```

Comment:
Pools are an organizing concept rather than a storage target so this is confusing here e.g. you don't write a file to a pool you write it to a container/object depending on which middleware is involved.